### PR TITLE
feat: add new module to autoconfigure grafana cloud

### PR DESCRIPTION
**Pull request by theSuess**
This PR adds a new module to simplify connecting to Grafana cloud by only having to specify the stack name and token.

### CODE REVIEW

**Review by mattdurham — APPROVED** [review]
This looks great. Will let @erikbaranowski also take a look.

File: modules/grafana-cloud/autoconfigure/module.river [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+export "logs_receiver" {
+  value = loki.write.default.receiver
+}
+
+export "profiles_receiver" {
+  value = pyroscope.write.default.receiver
+}
+export "stack_information" {
```
Comment (erikbaranowski) [inline modules/grafana-cloud/autoconfigure/module.river:28]: nit: spacing

File: modules/grafana-cloud/autoconfigure/module.river [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  value = json_decode(remote.http.config_file.content)
+}
+
+
+remote.http "config_file" {
```
Comment (erikbaranowski) [inline modules/grafana-cloud/autoconfigure/module.river:36]: nit: probably should have a new header to break it up from the Exports

File: modules/grafana-cloud/autoconfigure/module.river [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/********************************************
+ * ARGUMENTS
+ ********************************************/
+argument "stack_name" { }
+
+argument "token" {}
+
+
```
Comment (erikbaranowski) [inline modules/grafana-cloud/autoconfigure/module.river]: nit: extra space

File: modules/grafana-cloud/autoconfigure/module.river [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+/********************************************
+ * EXPORTS
```
Comment (erikbaranowski) [inline modules/grafana-cloud/autoconfigure/module.river:9]: nit: consistent order of MLTP with the Endpoints section below

File: modules/grafana-cloud/autoconfigure/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+* Access Policies need the `stacks:read` scope
+* API Keys need at least the the `MetricsPublisher` role
+
+## Module exports
+
+The following fields are exported by the module:
+
+| Name | Type | Description
+| ---- | ---- | -----------
+| `traces_receiver` | `otelcol.Consumer` | A value that other components can use to send trace data to.
```
Comment (erikbaranowski) [inline modules/grafana-cloud/autoconfigure/README.md]: similar ordering MLTP here probably

**Review by erikbaranowski — APPROVED** [review]
Great stuff., looks really good.

I added some nits on spacing and ordering but I'm happy to merge as is and do some non-functional tidying up later.

**Comment by theSuess** [comment]
I've fixed the things called out, should all be ready to merge now!

**Pull request merged**